public/pages/VisualCreatePolicy/components/UIActions/ShrinkUIAction.tsx

```diff
@@ -0,0 +1,278 @@
+          title="Force unsafe"
+          helpText={`If this is set to 'No' then the shrink action will fail for indices which do not have replica shards.`}
+          isInvalid={false}
+        />
+        <EuiFormRow fullWidth isInvalid={false} error={null} style={{ maxWidth: "100%" }}>
+          <EuiRadioGroup
+            options={radios}
+            idSelected={this.action.force_unsafe_input}
+            onChange={(id) => {
+              const forceUnsafe = id === "yes";
+              const shrinkObject = { ...action.action };
+              if (forceUnsafe) {
+                shrinkObject.shrink.force_unsafe = forceUnsafe;
+                shrinkObject.force_unsafe_input = id;
+              } else {
+                shrinkObject.shrink.force_unsafe = false;
+                shrinkObject.force_unsafe_input = "no";
+              }
+              onChangeAction(this.cloneUsingString(shrinkObject));
+            }}
+            name="forceUnsafe"
+          />
+        </EuiFormRow>
+        <EuiSpacer size="s" />
+        <EuiFormCustomLabel
+          title="Aliases"
+          helpText={`The aliases to be applied to the output shrunken index.`}
+          isInvalid={!this.isValidAliasesJson()}
+        />
+        <EuiFormRow fullWidth isInvalid={!this.isValidAliasesJson()} error={null} style={{ maxWidth: "100%" }}>
+          <DarkModeConsumer>
+            {(isDarkMode) => (
+              <EuiCodeEditor
+                mode="json"
+                theme={isDarkMode ? "sense-dark" : "github"}
+                width="100%"
+                value={action.action.aliases_json}
+                onChange={(str) => {
+                  const aliasesJSON = str;
+                  const shrinkObject: ShrinkAction = { ...action.action };
+                  // const shrink = { ...action.action.shrink };
+                  if (aliasesJSON) shrinkObject.aliases_json = aliasesJSON;
+                  else delete shrinkObject.aliases_json;
+                  onChangeAction(this.cloneUsingString(shrinkObject));
+                }}
+                setOptions={{ fontSize: "14px" }}
+                aria-label="Code Editor"
+                height="100px"
+              />
+            )}
+          </DarkModeConsumer>
+        </EuiFormRow>
+      </>
+    );
+  };
+
+  toAction = () => {
+    // Use the spread operator to copy the action
+    const shrink = { ...this.action };
+    if (shrink.aliases_json != null) {
+      shrink.shrink.aliases = JSON.parse(shrink.aliases_json as string);
+      console.log(shrink.aliases_json as string);
+      console.log((shrink.shrink.aliases as Object[])[0]);
+    }
+    if (shrink.target_index_name_template_json != null) {
+      shrink.shrink.target_index_name_template = JSON.parse(shrink.target_index_name_template_json as string);
+    }
+    delete shrink.aliases_json;
+    delete shrink.force_unsafe_input;
+    delete shrink.target_index_name_template_json;
+    console.log(shrink);
```

Comment:
remove?

Reply:
I'll remove the console.log (yikes thanks for the catch) but deleting the jsons seems to be the pattern that we use for parsing jsons into action objects and deleting these jsons from the action copy so these extra parameters aren't sent to the backend